src/data_source/storage/sql.rs

```diff
@@ -1170,6 +1192,401 @@ where
+                                        RETURNING id",
+                                        &[&writer.to_vec()],
+                                    )
+                                    .await?;
+
+                                children_values.push(r.get::<_, i32>(0));
+                                // Mark the entry as 1 in bitvec to indiciate a non-empty child
+                                children_bitvec.push(true);
+                            }
+                        }
+                    }
+
+                    // advance the traversal path for the internal nodes at each iteration
+                    // The final node would be the Root node where this iterator is exhausted
+                    traversal_path.next();
+                    // insert internal node
+                    txn.upsert(
+                        name,
+                        ["pos", "created", "hash_id", "children", "children_bitvec"],
+                        ["pos", "created"],
+                        [[
+                            sql_param(&LTree::from(traversal_path.clone())),
+                            sql_param(&created),
+                            sql_param(&hash_id),
+                            sql_param(&children_values),
+                            sql_param(&children_bitvec),
+                        ]],
+                    )
+                    .await?;
+                }
+
+                // We only care about the Branch Nodes here
+                // Leaf node is already inserted
+                _ => continue,
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl<Types: NodeType> MerklizedStateDataSource<Types> for SqlStorage {
+    type Error = QueryError;
+
+    /// Retreives a Merkle path from the database
+    async fn get_path<
+        E: Element + Send + DeserializeOwned,
+        I: Index + Send + ToTraversalPath<A> + DeserializeOwned,
+        A: Unsigned,
+        T: NodeValue + Send + CanonicalDeserialize,
+    >(
+        &self,
+        state_type: &'static str,
+        tree_height: usize,
+        header_state_commitment_field: &'static str,
+        snapshot: Snapshot<Types>,
+        key: serde_json::Value,
+    ) -> QueryResult<MerklePath<E, I, T>> {
+        let index: I = serde_json::from_value(key).map_err(ParseError::Serde)?;
+        // Get the traversal path of the index
+        let path = I::to_traversal_path(&index, tree_height)
+            .into_iter()
+            .map(|x| x as i64)
+            .collect::<Vec<_>>();
+
+        let created = match snapshot {
+            Snapshot::<Types>::Commit(commit) => {
+                let commit: &[u8; 32] = commit.as_ref();
+                // Get the block height using the merkle commitment.
+                let query = self
+                    .query_one(
+                        &format!(
+                            "SELECT n.created FROM {state_type} n INNER JOIN Header h on 
```

Comment:
I don't think a `JOIN` is necessary here, since `n.created` is the same as `h.height`